pytest_pylint.py

```diff
@@ -281,19 +286,36 @@ def setup(self):
 
     def runtest(self):
         """Check the pylint messages to see if any errors were reported."""
+        pylint_output_file = self.config.option.pylint_output_file
         reported_errors = []
         for error in self.session.pylint_messages.get(self.rel_path, []):
             if error.C in self.config.option.pylint_error_types:
                 reported_errors.append(
                     error.format(self._msg_format)
                 )
+
+                if pylint_output_file:
+                    with open(pylint_output_file, 'a') as file:
```

Comment:
Sorry but I hate opening and closing files like this, even though it's a test.
Can't we extract this loop into another method and do something like:
```
def _loop_errors(writer):
  for error in ...:
    # do something
    writer('error_writings_etc')

def runtest():
  if output_file:
    with open(...) as f:
      reported_errors = _loop_errors(writer=f.write)
  else:
    reported_errors = _loop_errors(writer=lambda *args, **kwargs: None)
...
```

Reply:
Yeah, it was very dumb code indeed.  Thanks for the good suggestion :+1: